newIDE/app/src/EventsSheet/ParameterFields/ObjectPointNameField.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import React, { Component } from 'react';
+import GenericExpressionField from './GenericExpressionField';
+import { type ParameterFieldProps } from './ParameterFieldCommons';
+import { type ExpressionAutocompletion } from '../../ExpressionAutocompletion';
+import getObjectByName from '../../Utils/GetObjectByName';
+import { getLastObjectParameterValue } from './ParameterMetadataTools';
+import { getAllPointNames } from '../../ObjectEditor/Editors/SpriteEditor/Utils/SpriteObjectHelper';
+
+const gd: libGDevelop = global.gd;
+
+export default class ObjectPointNameField extends Component<
+  ParameterFieldProps,
+  void
+> {
+  _field: ?GenericExpressionField;
+
+  focus() {
+    if (this._field) this._field.focus();
+  }
+
+  getPointNames(props: ParameterFieldProps): Array<ExpressionAutocompletion> {
+    const {
+      project,
+      scope,
+      instructionMetadata,
+      instruction,
+      expressionMetadata,
+      expression,
+      parameterIndex,
+    } = props;
+
+    const objectName = getLastObjectParameterValue({
+      instructionMetadata,
+      instruction,
+      expressionMetadata,
+      expression,
+      parameterIndex,
+    });
+    if (!objectName || !project) {
+      return [];
+    }
+
+    const object = getObjectByName(project, scope.layout, objectName);
+    if (!object) {
+      return [];
+    }
+
+    const spriteObject = gd.asSpriteObject(object);
+    if (!spriteObject) {
```

Comment:
Same here, check `getType` and avoid the early return to hint that this could be extended maybe in the future.